plugins/hu.elte.txtuml.api.model/src/hu/elte/txtuml/api/model/ModelExecutorThread.java

```diff
@@ -76,24 +79,20 @@ private void newMailboxEntry(MailboxEntry entry) {
 	 * becomes empty.
 	 */
 	void shutdown() {
-		try {
-			class ShutdownAction implements MailboxEntry {
-				@Override
-				public void execute() {
-					if (mailbox.isEmpty()) {
-						ModelExecutor.shutdownNow();
-					} else {
-						try {
-							mailbox.put(new ShutdownAction());
-						} catch (InterruptedException e) {
-						}
-					}
-				}
-			}
+		shutdownWhenEmpty = true;

```

Comment:
I think there is no problem because only the model executor thread reads it. Change will not appear immediately in that thread, but unsynchronized execution of shutdown cannot resolve in infinite loop (cannot set accidentally the already true flag to false).